### PR TITLE
[mlir][NFC] Mark type converter in `populate...` functions as `const`

### PR DESCRIPTION
**Pull request by matthias-springer**
This commit marks the type converter in `populate...` functions as `const`. This is useful for debugging.

Patterns already take a `const` type converter. However, some `populate...` functions do not only add new patterns, but also add additional type conversion rules. That makes it difficult to find the place where a type conversion was added in the code base. With this change, all `populate...` functions that only populate pattern now have a `const` type converter. Programmers can then conclude from the function signature that these functions do not register any new type conversion rules.

Also some minor cleanups around the 1:N dialect conversion infrastructure, which did not always pass the type converter as a `const` object internally.

### CODE REVIEW

File: mlir/lib/Conversion/TosaToTensor/TosaToTensor.cpp
Comment (zero9178) [inline mlir/lib/Conversion/TosaToTensor/TosaToTensor.cpp:1]: nit: This file seems to have some unrelated formatting changes that should/can probably be committed separately

**Review by zero9178 — APPROVED** [review]
LGTM!

**Review by kuhar — APPROVED** [review]
